projects/RabbitMQ.Client/client/impl/ConsumerDispatching/ConsumerDispatcher.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Buffers;
+using System.Threading.Tasks;
+using RabbitMQ.Client.Events;
+using RabbitMQ.Client.Impl;
+
+namespace RabbitMQ.Client.ConsumerDispatching
+{
+    #nullable enable
+    internal sealed class ConsumerDispatcher : ConsumerDispatcherChannelBase
```

Comment:
Is there any need for synchronous consumer? Why not remove it and just keep the async one? It can complete synchronously without allocations if it is ValueTask based.

Reply:
See here: https://devblogs.microsoft.com/dotnet/understanding-the-whys-whats-and-whens-of-valuetask/#valuetask-tresult-and-synchronous-completion

Reply:
This is something to be looked at for sure, but to not break the API further for the next release I kept it for now.

Reply:
Yeah, eventually it can be removed but not yet.